jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/EagerFormHandler.java

```diff
@@ -53,31 +52,74 @@ public boolean handle(Request request, org.eclipse.jetty.server.Response respons
         if (mimeType == null)
             return super.handle(request, response, callback);
 
-        CompletableFuture<?> future =  switch (mimeType)
+        return switch (mimeType)
         {
-            case FORM_ENCODED -> FormFields.from(request);
-            case MULTIPART_FORM_DATA -> ServletMultiPartFormData.from(Request.as(request, ServletContextRequest.class).getServletApiRequest(), contentType);
-            default -> null;
+            case FORM_ENCODED -> handleFormFields(request, response, callback);
+            case MULTIPART_FORM_DATA -> handleMultiPartFormData(request, contentType, response, callback);
+            default -> super.handle(request, response, callback);
         };
+    }
 
-        if (future == null)
-            return super.handle(request, response, callback);
+    protected boolean handleFormFields(Request request, org.eclipse.jetty.server.Response response, Callback callback)
+    {
+        Request.Handler handler = super::handle;
+        Promise<Fields> onFields = new Promise<>()
+        {
+            @Override
+            public void failed(Throwable x)
+            {
+                succeeded(null);
+            }
 
-        future.whenComplete((result, failure) ->
+            @Override
+            public void succeeded(Fields result)
+            {
+                try
+                {
+                    if (!handler.handle(request, response, callback))
+                        callback.failed(new IllegalStateException("Not Handled"));
+                }
+                catch (Throwable t)
+                {
+                    callback.failed(t);
+                }
+            }
+        };
+
+        InvocablePromise<Fields> executeOnFields = Invocable.from(request.getContext(), onFields);
+        FormFields.onFields(request, onFields, executeOnFields);
```

Comment:
@lorban but they do not.  the onFields is not executed and probably blocks.  The executedOnFields executes a call to onFields and does not block!

Reply:
I'm with @lorban here: this is an implementation detail and we should not provide a user API with 2 promises for the same result.
Before, we did not care.
We are deprecating the API, so an extra dispatch on a deprecated API should not be a problem.
We provide a new API based on Promise to `FormFields`.

This new API in `FormFields` with 2 promises is only used here.
We can easily replace it with a CountDownLatch here, and avoid the new API in `FormFields`.

Reply:
@sbordet @lorban The need for this feature is not for a deprecated API.  In fact it is for the most likely hot path when we are using handlers like `EagerFormHandler` or `DelayedHandler`.   We want to be able to load the content asynchronously, but in many (most?) cases, the content will come with the headers and already be there.  Thus we do not want to dispatch on the most common hot path case.

If you don't like the two Promise API, then we need another API that will allow the same behaviour.   We cannot have it that adding the `EagerFormHandler` or `DelayedHandler` slows down the best case requests.

An alternative API would be to pass a single Promise and an Executor that is used only if the result is not immediately available.   That is a little special purpose, but it is the only use-case we have for the two promises.

Reply:
@gregw this is orthogonal to this PR. Before we did not have such mechanism.
We do not need an API, it's just an implementation detail for only one Handler.

Reply:
It is not orthogonal to this PR. We've always had this behaviour and it was provided by CF. If we hide/remove CF then we need an API to the handlers that need it can do it. There is no easy way for a handler to do this without API

Reply:
@sbordet @lorban I've found a way to only have a single Promise. But it moves the future execution logic into the onFields style methods.